doc/source/loggers.rst

```diff
@@ -13,26 +13,133 @@ by caproto, below. Because recalling the name is not always convenient, certain
 objects in caproto's Python API expose a ``log`` attribute, such as ``pv.log``,
 ``pv.context.log`` and ``pv.circuit_manager.log``.
 
+You could find python logging flow here.
+https://docs.python.org/3/howto/logging.html#logging-flow
+
+
 Useful snippets
 ===============
 
-To turn on maximal logging (usually way too much information!) use
+To get the caproto logger by python built-in logging framework:
 
 .. code-block:: python
 
-    import logging
-    logging.getLogger('caproto').setLevel('DEBUG')
+   import logging
+   logger = logging.getLogger('caproto')
 
-To log (non-batch) read/write requests and read/write/event responses in the
-threading and pyepics-compat clients:
+To add a stdout handler with level:
 
 .. code-block:: python
 
-   import logging
-   logging.getLogger('caproto.ch').setLevel('DEBUG')
+    std_handler = logging.StreamHandler()
+    logger.addHandler(std_handler)
+
+To have a nicer stdout, we offer you a convenient way to setup. It summarize
+code above by one commmand:
+
+.. code-block:: python
+
+    from caproto._log import set_handler
+    set_handler(level='DEBUG')
+
+The output is like:
+
+.. code-block:: guess
+
+    [D 15:19:14.132          client:  588] [CLIENT] Broadcaster command loop is running.
+    [D 15:19:14.133          client:  719] [CLIENT] Broadcaster check for unresponsive servers loop is running.
+    [D 15:19:14.134          client:  454] [CLIENT] [0.0.0.0:0] --->>> [127.0.0.1:5065] Sending 16 bytes to 127.0.0.1:5065
+    [D 15:19:14.135          client:  454] [CLIENT] [0.0.0.0:0] --->>> [10.2.255.255:5065] Sending 16 bytes to 10.2.255.255:5065
+    [D 15:19:14.136          client:  832] [CLIENT] Broadcaster search-retry thread has started.
+    [D 15:19:14.136          client: 1079] [CLIENT] Context search-results processing loop has started.
+    [D 15:19:14.137          client:  872] [CLIENT] Sending 6 SearchRequests
+
+To add logstash handler, you can submit logs to a centralized log database:
+
+.. code-block:: python
+
+    import logstash
+    logstash_handler = logstash.TCPLogstashHandler(<host>, <port>, version=1)
+    logger.addHandler(logstash_handler)
+
+To add a file handler:
+
+.. code-block:: python
+
+    file_handler = logging.FileHandler('caproto.log')
+    file_handler.setLevel('DEBUG')
+    logger.addHandler(file_filter)
+
+To add a filter:
+
+.. code-block:: python
+
+    from caproto._log import PVFilter, AddressFilter, RoleFilter
+    std_handler.addFilter(PVFilter(['complex:*', 'simple:B'], level='DEBUG', exclusive=False))
```

Comment:
You should break these examples into different sections and really drive home what's possible with these filters. They are very cool, but if we don't tell people how to use them no one will know they exist...

Reply:
There should also be an API section that shows the `autoclass` information from each of the filters.